### PR TITLE
App js refactoring using React Context closes #4

### LINKED ISSUE #4: App js refactoring
App.js has become a little complex and we need to make it a little simpler.
It doesn't need to hold the app state.
We can use the [React Context](https://reactjs.org/docs/context.html) for this reason.
Please **study** the above link and then create two (DevicesContext, PlacesContext) new react context components that will provide the state to whichever needs it. 
Then each component that needs access to this context will consume it and call methods in that context component.
All the state concerning devices and places in App.js can now be removed

### PR DESCRIPTION
**Pull request by gdagalakis**

### CODE REVIEW

File: src/components/DeviceForm/index.js [hunk 1]
```diff
@@ -31,28 +32,32 @@ class DeviceForm extends Component {
...
-                <FormInput
-                    desc="Description"
-                    name="description"
-                    value={form.description || ''}
-                    onChange={this.handleChange('description')}
-                />
-                <input type="submit" value="Submit" />
-            </FormWrapper>
+            <DevicesContext.Consumer>
+                {value => (
```
Comment (nikoskleidis) [inline src/components/DeviceForm/index.js]: If you only use onSubmit from here, then it's better that you destructure it directly from value

File: src/components/PlaceForm/index.js [hunk 2]
```diff
@@ -30,38 +30,42 @@ class PlaceForm extends Component {
...
-                        desc="Long"
-                        name="posLong"
-                        value={form.posLong || ''}
-                        onChange={this.handleChange('posLong')}
-                    />
-                </div>
-                <input type="submit" value="Submit" />
-            </FormWrapper>
+            <PlacesContext.Consumer>
+                {value => (
```
Comment (nikoskleidis) [inline src/components/PlaceForm/index.js]: here as well

File: src/pages/Devices/index.js [hunk 3]
```diff
@@ -1,8 +1,14 @@
 import React from 'react'
 import DeviceList from '../../components/DeviceList'
-
-const Devices = props => {
-    return <DeviceList {...props} />
+import { DevicesContext } from 'lib/DevicesProvider'
+const Devices = () => {
+    return (
+        <div>
```
Comment (nikoskleidis) [inline src/pages/Devices/index.js]: Is the div really needed here or can we directly return the DevicesContext.Consumer?

**Review by nikoskleidis — CHANGES_REQUESTED** [review]

File: src/pages/Devices/index.js (hunk 3 above)
Comment (gdagalakis) [reply src/pages/Devices/index.js]: Removed, it was a leftover

**Pull request merged**